src/better-shifting.cow

```diff
@@ -422,10 +422,191 @@ or by a scalar shift amount which applies to all elements in the \tcode{simd::ve
+\h3{[simd]}
+
+In \eelis{simd.syn}, change the synopsis as follows:
+
+\Bdiff{
+\itemdecl{
+\html{[\N{HORIZONTAL ELLIPSIS}]}
+
+// \serif{[simd.bit], Bit manipulation}
+
+\html{[\N{HORIZONTAL ELLIPSIS}]}
+
+\ins{
+template<\exposid{simd-type} V0, \exposid{simd-type} V1>
+  constexpr V0 shl(const V0& v, const V1& s) noexcept;
+template<\exposid{simd-type} V>
+  constexpr V  shl(const V& v, int s) noexcept;
+
+template<\exposid{simd-type} V0, \exposid{simd-type} V1>
+  constexpr V0 shr(const V0& v, const V1& s) noexcept;
+template<\exposid{simd-type} V>
+  constexpr V  shr(const V& v, int s) noexcept;
+}
+
+template<\exposid{simd-type} V0, \exposid{simd-type} V1>
+  constexpr V0 rotl(const V0& v, const V1& s) noexcept;
+template<\exposid{simd-type} V>
+  constexpr V  rotl(const V& v, int s) noexcept;
+
+template<\exposid{simd-type} V0, \exposid{simd-type} V1>
+  constexpr V0 rotr(const V0& v, const V1& s) noexcept;
+template<\exposid{simd-type} V>
+  constexpr V  rotr(const V& v, int s) noexcept;
+
+\html{[\N{HORIZONTAL ELLIPSIS}]}
+
+// See \serif{[simd.bit], Bit manipulation}
+using simd::byteswap;
+using simd::bit_ceil;
+using simd::bit_floor;
+using simd::has_single_bit;
+\ins{using simd::shl;}
+\ins{using simd::shr;}
+using simd::rotl;
+using simd::rotr;
+using simd::bit_width;
+using simd::countl_zero;
+using simd::countl_one;
+using simd::countr_zero;
+using simd::countr_one;
+using simd::popcount;
+
+\html{[\N{HORIZONTAL ELLIPSIS}]}
+}
+}
+
+In \eelis{simd.bit}, immediately preceding the first declaration of
+\tcode{rotl}, insert the following:
+
+\Bins{
+\itemdecl{
+template<\exposid{simd-type} V0, \exposid{simd-type} V1>
+  constexpr V0 shl(const V0& v0, const V1& v1) noexcept;
+template<\exposid{simd-type} V0, \exposid{simd-type} V1>
+  constexpr V0 shr(const V0& v0, const V1& v1) noexcept;
+}
+
+\itemdescr{
+\constraints
+\ul{
+  \li{The type \tcode{V0::value_type} is a signed or unsigned integer type
+      \iref{basic.fundamental},}
+  \li{the type \tcode{V1::value_type} models \tcode{integral},}
+  \li{\tcode{V0::size() == V1.size()} is \tcode{true}, and}
+  \li{\tcode{sizeof(typename V0::value_type) == sizeof(typename V1::value_type)}
+      is \tcode{true}.}
+}
+
+\returns
+A \tcode{basic_simd} object where the \math{\msup{\mi{i}\mtext{th}}} element
+is initialized to the result of \tcode{\exposid{bit-func}(v0[\math{\mi{i}}],
+static_cast<int>(v1[\math{\mi{i}}]))} for all \math{\mi{i}} in the range
+[\tcode{0}, \tcode{V::size()}), where \tcode{\exposid{bit-func}} has the same
+behavior as the corresponding scalar function from \tcode{<bit>} except that the
+type of its second parameter is considered to be \tcode{V1::value_type}.
+}
+
+\itemdecl{
+template<\exposid{simd-type} V>
+  constexpr V shl(const V& v, int s) noexcept;
+template<\exposid{simd-type} V>
+  constexpr V shl(const V& v, int s) noexcept;
+}
+
+\itemdescr{
+\constraints
+The type \tcode{V::value_type} is a signed or unsigned integer type
+\iref{basic.fundamental}.
```

Comment:
```suggestion
The type \tcode{V::value_type} is a signed or unsigned integer type\iref{basic.fundamental}.
```
IIRC `\iref` expands to a leading non-breaking space, and that won't work if it's not tight against the word on the left, like in the standard draft.